### PR TITLE
Implement Progress List Feature with Comprehensive Tests (Issues #85)

### PR DESCRIPTION
**Pull request by Zumh**
$ lms progress list
+----+---------------+--------------+------------+
| Id |  Description  |   WorkItem   | CreatedAt  |
+----+---------------+--------------+------------+
| p0 | MyDescription | Assignment 2 | 2024-10-01 |
+----+---------------+--------------+------------+

## Summary
Description: This pull request introduces the “progress list” feature, addressing the requirements outlined in Issue #85. The feature allows users to view their progress in a structured and user-friendly format.

## Key Changes:
Added the **DisplayProgressSummary** component to display user progress.
Implemented the **GetProgresses** method to retrieve progress data.
Created the **GetDisplayProgressSummary** method to format and present progress information.
Included comprehensive unit tests to ensure the functionality and reliability of the new feature.
## Testing:

Verified that the **DisplayProgressSummary** component correctly displays progress data.
Ensured that edge cases, such as empty progress lists, are handled gracefully.
All tests pass successfully, confirming the robustness of the implementation.

## Related Issue:
#85

### CODE REVIEW

File: app/Properties/launchSettings.json
Comment (ehharvey) [inline app/Properties/launchSettings.json:5]: I'm approving this for now, but we should avoid using this approach moving forward. I'm in favour of using the `.vscode/launch.json` file See #216

File: app/Progresses.cs
Comment (ehharvey) [inline app/Progresses.cs:29]: We should remove this.
```suggestion
```

File: app/Progresses.cs
Comment (ehharvey) [inline app/Progresses.cs:37]:
```suggestion
```

File: app/Progresses.cs
Comment (ehharvey) [inline app/Progresses.cs:80]: Nice :)

File: app/Progresses.cs
Comment (ehharvey) [inline app/Progresses.cs:105]: Let's fix this to not use temp data!!

File: app/Progresses.cs
Comment (ehharvey) [inline app/Progresses.cs:146]:
```suggestion
```

File: app/Progresses.cs
Comment (ehharvey) [inline app/Progresses.cs:150]:
```suggestion
}
```

File: app-test/ProgressTests.cs
Comment (ehharvey) [inline app-test/ProgressTests.cs:82]: I would like to see some tests that create a new progress item and verify that the `GetProgresses()` works as expected. See my PR #215 for an example :)

Comment (ehharvey) [reply app-test/ProgressTests.cs:82]: Part of this is some issue with the DB persisting between DBs. The ^^ PR has a fix for this. It is *tricky* so all good if it's hard to figure out. I can copy my changes over if needed!

**Review by ehharvey — CHANGES_REQUESTED** [review]
Overall, looks good! There a few small changes to make. Mainly, I wanna see the `GetProgresses` fetch from the DB and the tests check that it does so properly. See my PR #215 for an example!

Part of the issue is that I didn't know about the DB being crazy in the test project. My branch has a fix for that. I'm working on the other work item related work, then will merge all of that. After I do so, I'll merge those changes onto this branch! That will give you some help :)

**Comment by ehharvey** [comment]
Hi again :) I've just finished up my `WorkItem` related issues. What I'll do is merge those in, then pull those changes to your PR. I'm hoping that'll give a better sense of what is required :)

**Comment by ehharvey** [comment]
> Hi again :) I've just finished up my `WorkItem` related issues. What I'll do is merge those in, then pull those changes to your PR. I'm hoping that'll give a better sense of what is required :)

Done!

**Comment by ehharvey** [comment]
@Zumh I believe I have already mentioned to not make forks of the repository :/ Can you reply here to confirm? Thanks!

**Comment by Zumh** [comment]
@ehharvey sorry was mia. I'll fix this right away.

File: app/Properties/launchSettings.json
Comment (Zumh) [reply app/Properties/launchSettings.json:5]: Yeah I'm using Visual studio 22 which everyone recommended.

**Comment by ehharvey** [comment]
@Zumh no worries :)

Generally, when you create pull requests, there will always be a bit of revision needed. I know sometimes it can be awkward to share your work and get feedback for revisions, but I promise it is all worth it in the end :) I'd say it's more unusual to _not_to revisions.

I saw you removed the link to #85 . Was this intended?

**Comment by Zumh** [comment]
@ehharvey I copy the code over to a new branch that I created and removed link to this pull request.

**Comment by ehharvey** [comment]
Ok! Create a new PR when you are ready. You _can_ just use this PR if you'd like (e.g., reopen and address the comments from ^^). The comment about using branches is mostly just to make sure we're aligned in processes :) So it is something I'd prefer moving forward.

That said, I'll leave it to you whether you wanna use the new branch or not.

The feedback I created _is_ on this PR. Definitely try to address as much as possible :) If/when you create the new PR, I will review what's there :)

**Comment by ehharvey** [comment]
@Zumh I wasn't able to find the new branch? Can you pass it over? (but also see my comment above; I'm happy to keep the work using this fork + PR)

**Comment by ehharvey** [comment]
ooops, looks like its literally `85-view-progresses` lol :facepalm: .

**Pull request closed without merging**
